module/VuFind/src/VuFind/Db/Table/ExpirationTrait.php

```diff
@@ -49,61 +48,51 @@ trait ExpirationTrait
      *
      * @param Select $select  Select clause
      * @param int    $daysOld Age in days of an "expired" record.
-     * @param int    $idFrom  Lowest id of rows to delete.
-     * @param int    $idTo    Highest id of rows to delete.
      *
      * @return void
      */
-    abstract protected function expirationCallback($select, $daysOld, $idFrom = null,
-        $idTo = null
-    );
+    abstract protected function expirationCallback($select, $daysOld);
 
     /**
      * Delete expired records. Allows setting of 'from' and 'to' ID's so that rows
      * can be deleted in small batches.
      *
-     * @param int $daysOld Age in days of an "expired" record.
-     * @param int $idFrom  Lowest id of rows to delete.
-     * @param int $idTo    Highest id of rows to delete.
+     * @param int      $daysOld Age in days of an "expired" record.
+     * @param int|null $limit   Maximum number of rows to delete or null for no
+     * limit.
      *
      * @return int Number of rows deleted
      */
-    public function deleteExpired($daysOld = 2, $idFrom = null, $idTo = null)
+    public function deleteExpired($daysOld, $limit = null)
     {
-        // Determine the expiration date:
-        $callback = function ($select) use ($daysOld, $idFrom, $idTo) {
-            $this->expirationCallback($select, $daysOld, $idFrom, $idTo);
+        // Determine the expiration parameters:
+        $lastId = $limit ? $this->getExpiredBatchLastId($daysOld, $limit) : null;
+        $callback = function ($select) use ($daysOld, $lastId) {
+            $this->expirationCallback($select, $daysOld, $lastId);
```

Comment:
Why are you passing $lastId here when expirationCallback() only appears to accept two parameters?

Reply:
Oops, that's a leftover from moving the $lastId handling from the callback to the trait. I'll remove it.